src/textCampaign/textCampaign.controller.ts

```diff
@@ -1,29 +1,154 @@
+import { PublicAccess } from 'src/authentication/decorators/PublicAccess.decorator'
 
-@Controller('text-campaign')
+@Controller('text-campaigns')
+@UsePipes(ZodValidationPipe)
 export class TextCampaignController {
-  constructor(private readonly textCampaignService: TextCampaignService) {}
+  private readonly logger = new Logger(TextCampaignController.name)
 
-  @Post('project')
+  constructor(
+    private readonly textCampaignService: TextCampaignService,
+    private readonly campaigns: CampaignsService,
+  ) {}
+
+  @Post()
   @UseCampaign()
-  @UseGuards(ValidCampaignGuard)
   createProject(
     @ReqCampaign() campaign: Campaign,
     @Body() createProjectDto: CreateProjectSchema,
   ) {
     return this.textCampaignService.createProject(campaign.id, createProjectDto)
   }
 
-  @Get('text-campaigns')
+  @Get()
   @UseCampaign()
   findAll(@ReqCampaign() campaign: Campaign) {
     return this.textCampaignService.findByCampaignId(campaign.id)
   }
+
+  @Post('compliance')
+  @UseCampaign()
+  async submitComplianceForm(
+    @ReqCampaign() campaign: Campaign,
+    @Body() body: ComplianceFormSchema,
+  ) {
+    const updatedCampaign = await this.campaigns.update({
+      where: { id: campaign.id },
+      data: {
+        data: {
+          ...campaign.data,
+          tcrComplianceInfo: { ...body, status: TcrComplianceStatus.submitted },
+        },
+      },
+    })
+
+    try {
+      this.logger.debug(
+        `Submitting compliance form for campaign ${campaign.id}`,
+        body,
+      )
+      await this.textCampaignService.submitComplianceForm(updatedCampaign, body)
+    } catch (e) {
+      this.logger.error(
+        `Failed to submit compliance form for campaign ${campaign.id}`,
+        e,
+      )
+      throw e
+    }
+
+    return updatedCampaign
+  }
+
+  // TODO: to be used for UI to submit pin
+  @Post('compliance/pin')
+  @UseCampaign()
+  async submitCompliancePin(
+    @ReqCampaign() campaign: Campaign,
+    @Body() { pin }: CompliancePinSchema,
+  ) {
+    await this.campaigns.update({
+      where: { id: campaign.id },
+      data: {
+        data: {
+          tcrComplianceInfo: {
+            ...(campaign.data.tcrComplianceInfo as TcrComplianceInfo),
+            pin,
+            status: TcrComplianceStatus.pending,
+          },
+        },
+        ...campaign.data,
+      },
+    })
+    return this.textCampaignService.submitCompliancePin(campaign, pin)
+  }
+
+  // TODO: to be used for webhook from RumbleUp
+  @Post('compliance/approve')
+  @PublicAccess()
+  async approveCompliance(
+    // TODO: what will the payload actually be?
+    @Body() body: { campaignId: number; approved: boolean },
+  ) {
+    const status = body.approved
+      ? TcrComplianceStatus.approved
+      : TcrComplianceStatus.rejected
+
+    this.logger.debug(
+      `Received TCR compliance approval for Campaign: ${body.campaignId}, Status: ${status}`,
+    )
+    // TODO: how will we know what campaign to update?
```

Comment:
This will be based off the TCR `campaignId` that we _should_ receive when we first submit the TCR info for the start of the process.. 

https://app.rumbleup.com/app/docs/api-projects#:~:text=optional)%3A%20CSV%20file%20containing%20group%20information%20(if%20group%20is%20%E2%80%9Cnull%E2%80%9D)-,campaignId%20(optional)%3A%20TCR%20campaign%20Id%20for%20the%20action.%20See%20TCR%20Campaign%20Select.%20This%20replaces%20depricated%20tcr_cid%20parameter.,-proxy%20(optional)%3A%20If%20TCR%20campaignId%20is%20provided%20then%20this%20must

i.e. even tho we don't have approval as of yet, we do already have a TCR campaign ID: 

<img width="590" alt="image" src="https://github.com/user-attachments/assets/69fde47d-823a-4865-8730-2bee7622fecc" />